### PR TITLE
Arrow/Bolt changes

### PR DESCRIPTION
**Pull request by darktoes**
Changes metal/steel/carbon fibre arrows and bolts to be a bit more realistic, and to tie in with later changes to weapons that use bolts as ammunition. Minor adjustments to range/damage etc. Moves their dispersion a bit closer together, and makes them far more likely to be recovered.

Also changes standard metal/steel arrows/bolts to be broadhead (standard cutting blade). Adds bodkin (armor piercing) and target (accuracy/range bonus) variants for metal/steel arrows/bolts.

### CODE REVIEW

File: data/json/items/archery.json [hunk 1]
```diff
@@ -184,32 +184,76 @@
...
+	  "effects": ["RECOVER_60"]
+   }, 
    {
       "type" : "AMMO",
       "id" : "arrow_cf",
       "price" : 9500,
       "name" : "carbon fiber arrow",
       "symbol" : "=",
       "color" : "green",
-      "description" : "High-tech carbon fiber shafts and 100 grain broadheads.  Very lightweight, fast, and notoriously fragile.  Stands a bad chance of remaining intact once fired.",
```
Comment (DracoGriffin) [inline data/json/items/archery.json]: Spaces between `"description" :` should be reverted for styling. Similarly, two spaces between sentences (this should be in [JSON_STYLE.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_STYLE.md) but looks like that's missing), not just one.

File: data/json/items/archery.json [hunk 2]
```diff
@@ -184,32 +184,76 @@
...
-      "dispersion" : 135,
+      "damage" : 16,
+      "pierce" : 8,
+      "range" : 14,
+      "dispersion" : 65,
       "loudness": 0,
       "count" : 8,
       "stack_size" : 10,
-      "effects" : ["RECOVER_3"]
+      "effects" : ["RECOVER_80"]
```
Comment (DracoGriffin) [inline data/json/items/archery.json]: What type of carbon fiber is this based off of? [This source](https://onlinelibrary.wiley.com/doi/full/10.1111/1556-4029.13503) indicates most carbon fiber arrows are quite fragile.
>Carbon fiber arrows have smaller diameters than their aluminum arrow counterparts due to the strength of this material. Their light weight allows for higher velocities and for longer travel distances. The disadvantage of these arrows is that the carbon fiber can be fragile and is not subject to reuse after damage

File: data/json/items/crossbows.json [hunk 3]
```diff
@@ -44,23 +44,67 @@
       "type" : "AMMO",
       "id" : "bolt_steel",
       "price" : 5000, "//" : "That's the going rate for aluminum, but unless you want carbon fiber, that's as metal as Cabela's seems to stock.",
-      "name" : "steel crossbow bolt",
```
Comment (DracoGriffin) [inline data/json/items/crossbows.json]: Similar style/format issues as above.

File: data/json/items/crossbows.json [hunk 4]
```diff
@@ -44,23 +44,67 @@
...
+	  "description": "A sharp bolt made from steel with extra long fletching and a round, balanced tip for extra accuracy, at the cost of some damage. Deadly in skilled hands. Stands an excellent chance of remaining intact once fired.",
+	  "material": "steel",
+	  "volume": 1,
+	  "weight": 122,
+	  "bashing": 3,
+	  "cutting": 2,
+	  "ammo_type": "bolt",
+	  "damage": 10,
+	  "pierce": 7,
+	  "range": 12,
```
Comment (DracoGriffin) [inline data/json/items/crossbows.json]: Again, is the target variants supposed to have higher or lower ranges than others?

>Adds bodkin (armor piercing) and target (accuracy/range bonus) variants

File: data/json/items/archery.json [hunk 5]
```diff
@@ -184,32 +184,76 @@
...
+	  "description": "This metal arrow has fletching and a steel bodkin arrowhead for armor penetration.  Stands a very good chance of remaining intact once fired.",
+	  "material": "steel",
+	  "volume": 1,
+	  "weight": 60,
+	  "bashing": 3,
+	  "cutting": 2,
+	  "ammo_type": "arrow",
+	  "damage": 10,
+	  "pierce": 15,
+	  "range": 12,
```
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/items/archery.json]: Whoops. Will fix.

File: data/json/items/archery.json [hunk 6]
```diff
@@ -184,32 +184,76 @@
       "damage" : 14,
       "pierce" : 7,
       "range" : 10,
-      "dispersion" : 60,
+	  "dispersion": 80,
```
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/items/archery.json]: The Linter I used is responsible, I'll try to fix it.

File: data/json/items/archery.json (hunk 1 above)
Comment (darktoes) [reply data/json/items/archery.json]: Will do. Weird though.

File: data/json/items/archery.json (hunk 2 above)
Comment (darktoes) [reply data/json/items/archery.json]: In my experience CF arrows are far more durable. The main difference is that metal arrows will deform and can sometimes be fixed while CF will just shatter or sometime chip. It takes significantly more effort to damage a CF arrow over a metal one though.

Not sure what to say on diameter. In my experience most target shooters use thin metal or CF arrows, while hunters use almost exclusively thicker CF.

File: data/json/items/archery.json [hunk 7]
```diff
@@ -171,7 +171,7 @@
       "type" : "AMMO",
       "id" : "arrow_metal",
       "price" : 5000,
-      "name" : "metal arrow",
+	  "name": "metal broadhead arrow",
```
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/items/archery.json]: Fixed.

File: data/json/items/archery.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/items/archery.json]: Fixed.

File: data/json/items/archery.json [hunk 8]
```diff
@@ -184,32 +184,76 @@
       "damage" : 14,
       "pierce" : 7,
       "range" : 10,
-      "dispersion" : 60,
+	  "dispersion": 80,
       "loudness": 0,
       "count" : 10,
-      "effects" : ["RECOVER_15"]
+	  "effects": ["RECOVER_50"]
```
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/items/archery.json]: Fixed.

File: data/json/items/archery.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/items/archery.json]: Fixed.

File: data/json/items/archery.json [hunk 9]
```diff
@@ -184,32 +184,76 @@
...
+	  "description": "This metal arrow has extra long fletching and a round, balanced arrowhead for extra accuracy, at the cost of some damage.  Stands a very good chance of remaining intact once fired.",
+      "material": "steel",
+	  "volume": 1,
+	  "weight": 50,
+	  "bashing": 3,
+	  "cutting": 2,
+	  "ammo_type": "arrow",
+	  "damage": 12,
+	  "pierce": 5,
+	  "range": 8,
```
[the remark this answers is not in the document]
Comment (darktoes) [reply data/json/items/archery.json]: Fixed.

File: data/json/items/archery.json (hunk 1 above)
Comment (darktoes) [reply data/json/items/archery.json]: Fixed.

File: data/json/items/crossbows.json (hunk 3 above)
Comment (darktoes) [reply data/json/items/crossbows.json]: Fixed.

File: data/json/items/crossbows.json (hunk 4 above)
Comment (darktoes) [reply data/json/items/crossbows.json]: Fixed.

**Comment by darktoes** [comment]
Added recipes for the new bolts. Whoops.

File: data/json/recipes/recipe_ammo.json [hunk 10]
```diff
@@ -575,6 +575,78 @@
...
+    [
+      [ "adhesive", 1, "LIST" ],
+      [ "filament", 20, "LIST" ]
+    ],
+    [
+      [ "scrap", 9 ],
+      [ "steel_chunk", 3 ]
+    ],
+    [
+      [ "feather", 10 ],
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_ammo.json]: I suggest creating a new `fletching` requirement list and replacing it in all arrows recipes.

Comment (darktoes) [reply data/json/recipes/recipe_ammo.json]: That's not a bad idea. I'll look into it.

Comment (darktoes) [reply data/json/recipes/recipe_ammo.json]: There, I put it in.

**Comment by darktoes** [comment]
The fletching requirement isn't working, no idea why. If anybody has any ideas, let me know.

File: data/json/requirements/materials.json [hunk 11]
```diff
@@ -86,5 +86,23 @@
     "type": "requirement",
     "//": "Materials for waterproofing leather and wooden items.",
     "components": [ [ [ "wax", 1 ], [ "pine_bough", 5 ] ] ]
+  },
+  {
+    "id": "fletching",
+    "type": "requirement",
+    "//": "Materials for fletching arrows.",
+    "components": [
+    [
```
Comment (Night-Pryanik) [inline data/json/requirements/materials.json]: There should be an indentation of two spaces from the vertical line of `"components"`.

```
"components": [
  [
    [ "feather", 10 ],
    ...
  ]
]
```

Comment (darktoes) [reply data/json/requirements/materials.json]: Fixed. Think that was causing it to fail?

File: data/json/items/crossbows.json [hunk 12]
```diff
@@ -44,23 +44,67 @@
...
+      "effects" : ["RECOVER_50"]
    },
+   {
+      "type": "AMMO",
+      "id": "bolt_steel_bodkin",
+      "price": 4000,
+      "name": "steel bodkin bolt",
+      "symbol": "=",
+      "color": "green",
+      "description": "A sharp bolt made from steel with an armor piercing bodkin tip.  Deadly in skilled hands. Stands an excellent chance of remaining intact once fired.",
```
Comment (ZhilkinSerg) [inline data/json/items/crossbows.json]: Two whitespaces between sentences here.

File: data/json/items/crossbows.json [hunk 13]
```diff
@@ -44,23 +44,67 @@
...
+      "effects": ["RECOVER_50"]
+   },
+   {
+      "type": "AMMO",
+      "id": "bolt_steel_target",
+      "price": 4000,
+      "name": "steel target bolt",
+      "symbol": "=",
+      "color": "green",
+      "description": "A sharp bolt made from steel with extra long fletching and a round, balanced tip for extra accuracy, at the cost of some damage. Deadly in skilled hands. Stands an excellent chance of remaining intact once fired.",
```
Comment (ZhilkinSerg) [inline data/json/items/crossbows.json]: Two whitespaces between sentences here too.

Comment (darktoes) [reply data/json/items/crossbows.json]: Both fixed.

**Pull request merged**